materialize-sql/type_mapping.go

```diff
@@ -475,12 +475,19 @@ func (c constrainter) compatibleType(existing boilerplate.EndpointField, propose
 
 func (c constrainter) migratable(existing boilerplate.EndpointField, proposed *pf.Projection, rawFieldConfig json.RawMessage) (bool, *MigrationSpec, error) {
 	proj := buildProjection(proposed, rawFieldConfig)
+	// We check migratability against nullable DDLs, as we also cast
+	// the type to the nullable form of the DDL, however the final new column will be
+	// created as non-nullable if the proposed projection requests it
+	if proj.Inference.Exists == pf.Inference_MUST {
```

Comment:
`mapped` below already has both the `DDL` and `NullableDDL`. Instead of doing this re-assignment to always seeintg `Inference_MAY`, can we pass `mapped.NullableDDL` in instead of `mapped.DDL` on line 483?

Reply:
ah yes, I wrote this before I realised we have `NullableDDL`, makes sense 👍🏽